src/device.rs

```diff
@@ -110,6 +110,32 @@ impl<'a> Device<'a> {
         }
     }
 
+    /// Returns the first parent with a given subsystem and devtype
+    pub fn parent_with_subsystem_devtype<T: AsRef<OsStr>, U: AsRef<OsStr>>(&self, subsystem: T, devtype: Option<U>) -> ::Result<Option<Device>> {
```

Comment:
I wonder if we can simplify the return type. I understand why it's `Result<Option<...>>`, but it doesn't feel ergonomic to handle that return type:

```rust
// this feels awkward
if let Ok(Some(parent)) = device.parent_with_subsystem("foo") {
    // ...
}
```

The only `Err` is if one of the arguments isn't a valid C-style string. If that's the case, I think the function's semantics would still be consistent with `None` as the return value.

```rust
device.parent_with_subsystem("\x00"); // None makes sense. There will be no subsystems
                                      // named "\x00", because that's not a valid C string.
```